internal/indexer/preloader.go

```diff
@@ -83,17 +84,23 @@ func visit(
 ) {
 	newPath := updateNodePath(path, node)
 	newMonikerPath := updateMonikerPath(monikerPath, node)
+	start := sort.Search(len(positions), func(i int) bool {
+		return positions[i] >= node.Pos()
+	})
+
+	end := start
+	for end < len(positions) && positions[end] <= node.End() {
```

Comment:
perf nit since it's a perf PR: 
finding end would be faster using another application of sort.Search, but idk how many positions an ast node can span for whether it matters

Reply:
My assumption is that it's fairly small and I don't want to break cache locality here.

Reply:
makes sense